### PR TITLE
[13.0][MIG] account_invoice_view_payment

### PR DESCRIPTION
**Pull request by kongrattapong**
Ready to review

### CODE REVIEW

File: account_invoice_view_payment/wizards/account_register_payments.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        payment_model = self.env["account.payment"]
+        payments = payment_model
+        for payment_vals in self.get_payments_vals():
+            payments += payment_model.create(payment_vals)
+        payments.post()
+        res = {
+            "domain": [("id", "in", payments.ids), ("state", "=", "posted")],
+            "views": [(False, "form")],
+            "name": _("Payments"),
+            "view_type": "form",
```
Comment (nikul-serpentcs) [inline account_invoice_view_payment/wizards/account_register_payments.py]: @kongrattapong FYI `view_type` is deprecated.

Comment (kongrattapong) [reply account_invoice_view_payment/wizards/account_register_payments.py]: @nikul-serpentcs Thank for review, I'll fix it.

File: account_invoice_view_payment/models/account_payment.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        return res
+
+
+class AccountPaymentRegister(models.TransientModel):
+    _inherit = "account.payment.register"
+
+    def create_payment_and_open(self):
+        payment_model = self.env["account.payment"]
+        payments = payment_model
+        for payment_vals in self.get_payments_vals():
```
Comment (kittiu) [inline account_invoice_view_payment/models/account_payment.py:31]: Can we create in batch, no need to for loop?
(I think v13 is capable of create in batch).

File: account_invoice_view_payment/models/account_move.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        for rec in reconciles:
+            payment.append(rec["account_payment_id"])
+
+        # choose the view_mode accordingly
+        if len(reconciles) != 1:
+            result["domain"] = "[('id', 'in', " + str(payment) + ")]"
+        else:
+            res = self.env.ref("account.view_account_payment_form", False)
+            result["views"] = [(res and res.id or False, "form")]
+            result["res_id"] = payment[0]
```
Comment (kittiu) [inline account_invoice_view_payment/models/account_move.py:33]:
```suggestion
            result["res_id"] = payment and payment[0] or False
```

**Review by kittiu — APPROVED** [review]
Some comments

**Pull request closed without merging**